backend/roles/team-roles-manager.js

```diff
@@ -2,45 +2,62 @@
 
 const twitchApi = require("../twitch-api/api");
 const frontendCommunicator = require("../common/frontend-communicator");
+const accountAccess = require("../common/account-access");
 
 /**
- * @param {import('twitch/lib/API/Kraken/Team/Team').Team[]} teams
+ * @typedef MappedTeamRole
+ * @property {string} id
+ * @property {string} name
  */
-function mapRoles(teams) {
+
+/** @returns {MappedTeamRole} */
+const mapRoles = (teams) => {
     return teams
         .map(team => {
             return {
-                id: team.id,
+                id: parseInt(team.id), // For compatibility (Kraken API returned an int)
                 name: team.displayName
             };
         });
-}
-
-async function getAllTeamRolesForViewer(username) {
-    const roles = await twitchApi.teams.getMatchingTeamsByName(username);
+};
 
-    return mapRoles(roles);
-}
+/** @returns {Promise.<MappedTeamRole[]>} */
+const getAllTeamRolesForViewer = async (username) => {
+    const client = twitchApi.getClient();
+    const user = await client.helix.users.getUserByName(username);
+    const streamerTeams = await client.helix.teams.getTeamsForBroadcaster(accountAccess.getAccounts().streamer.userId);
 
-async function getTeamRoles() {
-    const teams = await twitchApi.teams.getStreamerTeams();
+    if (streamerTeams == null) return null;
```

Comment:
Might make sense to just return an empty array here instead of null so any code that calls this function doesn't have to worry about null checks. That's typically my rule of thumb with functions that returns arrays.

Reply:
Fixed.